src/Build/TelemetryInfra/InternalTelemetryConsumingLogger.cs

```diff
@@ -75,6 +82,22 @@ private void FlushDataIntoConsoleIfRequested()
         Console.WriteLine("==========================================");
     }
 
+    private void FlushDataIntoJsonFileIfRequested()
+    {
+        const string jsonFileNameVariable = "MSBUILDNODETELEMETRYFILENAME";
+        var jsonFilePath = Environment.GetEnvironmentVariable(jsonFileNameVariable);
+        if (string.IsNullOrEmpty(jsonFilePath))
+        {
+            return;
+        }
+
+        var telemetryTags = _workerNodeTelemetryData.AsActivityDataHolder(true, true)?.GetActivityProperties();
```

Comment:
do we need to check `Traits.Instance.ExcludeTasksDetailsFromTelemetry` here?
could you also use param names? it's easier to read